personal_finances/bank_interface/disk_token_store.py

```diff
@@ -0,0 +1,73 @@
+from .token_store import TokenStore, TokenObject
+from typing import Optional
+import json
+from ..file_helper import write_string
+import logging
+from .token_store import TokenNotFound
+from pydantic import ValidationError
+import os
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TokenAbsolutePathNotConfigured(Exception):
+    pass
+
+
+class InvalidTokenAbsolutePath(Exception):
+    pass
+
+
+def _remove_trailing_char(input_string: str, trailing_char: str) -> str:
+    return input_string[:-1] if input_string[-1] == trailing_char else input_string
+
+
+def _contains_empty_path_part(absolute_path: str) -> bool:
+    return any(map(lambda path_part: path_part == "", absolute_path[1:].split("/")))
+
+
+def _get_path_from_env() -> str:
+    try:
+        return os.environ["TOKEN_DISK_PATH_PREFIX"]
+    except KeyError as e:
+        raise TokenAbsolutePathNotConfigured(e)
+
+
+def _get_token_disk_path_prefix() -> str:
+    configured_token_path = _get_path_from_env()
+
+    if not os.path.isabs(configured_token_path):
+        raise InvalidTokenAbsolutePath(f"is not absolute path: {configured_token_path}")
+
+    path_without_trailing_slash = _remove_trailing_char(configured_token_path, "/")
+
+    if _contains_empty_path_part(path_without_trailing_slash):
+        raise InvalidTokenAbsolutePath(
+            f"contains empty path parts: {configured_token_path}"
+        )
+
+    return path_without_trailing_slash
+
+
+class DiskTokenStore(TokenStore):
+    def save_token(self, user_id: str, token: TokenObject) -> Optional[TokenObject]:
+        previous_content = write_string(
+            f"{_get_token_disk_path_prefix()}/{user_id}.json",
+            token.model_dump_json(),
+        )
+        try:
+            previous_token = TokenObject.model_validate_json(previous_content)
```

Comment:
Why are you checking the previous token stored?

Reply:
Only because I'm returning it in this function, I need to return something valid.